x-pack/plugins/cases/server/client/cases/metrics/types.ts

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MetricsResponse } from '../../../../common';
+
+export interface MetricsHandler {
+  getFeatures(): Set<string>;
+  applyMetrics(metricsResult: MetricsResponse): Promise<MetricsResponse>;
```

Comment:
`applyMetrics` is a bit confusing to me. This function computes/calculates the metric. I think `compute` or similar is more appropriate. Also, I do not think it should be dependent on the `metricsResult` argument. That creates a tight relationship between the different metric handlers, gives more responsibility to the handlers that they should have, and creates problems if some handler mutates the `metricsResult` object. I think that the concatenation of the results should be handled by the `getMetrics` in `x-pack/plugins/cases/server/client/cases/metrics/index.ts`.